HACKING.md

```diff
@@ -0,0 +1,409 @@
+# Developing Salt
+
+
+So you want to contribute to the Salt project? Excellent! You can help in a
+number of ways:
+
```

Comment:
NOTE: I use the [conventional comments method](https://conventionalcomments.org/) of PR reviewing.

**Chore:** Let's define what contributing means and what kinds of contributions we're looking for. I'm a big fan of the [All Contributors Project ](https://github.com/all-contributors/all-contributors) statement that advocates for "recognizing contributors to an open source project in a way that rewards each and every contribution, not just code."

I don't think this is actually on you to do this in this PR, which is why I marked it as a follow-up chore. We need to maybe need to add this to the Open agenda and discuss it and then document the decision here. We could even go so far as considering adding the All Contributor's bot to our project, but that would require an SEP for sure. Could you help me open a ticket it or add it to the appropriate agenda and make sure I'm invited wherever that discussion occurs?